local-volume/bootstrapper/bootstrapper.go

```diff
@@ -0,0 +1,247 @@
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: provisionerNodeBindingName,
+		},
+		RoleRef: rbacv1beta1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     systemRoleNode,
+		},
+		Subjects: subjects,
+	}
+
+	_, err := client.RbacV1beta1().ClusterRoleBindings().Create(&pvBinding)
+	if err != nil {
+		return err
+	}
+	_, err = client.RbacV1beta1().ClusterRoleBindings().Create(&nodeBinding)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createDaemonSet(client *kubernetes.Clientset, namespace string, config map[string]common.MountConfig) error {
+	volumes := []v1.Volume{}
+	volumeMounts := []v1.VolumeMount{}
+	for _, mount := range config {
+		name := generateMountName(mount.HostDir, mount.MountDir)
+		volumes = append(volumes, v1.Volume{
+			Name: name,
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: mount.HostDir,
+				},
+			},
+		})
+		volumeMounts = append(volumeMounts, v1.VolumeMount{
+			Name:      name,
+			MountPath: mount.MountDir,
+		})
+	}
+
+	envVars := []v1.EnvVar{
+		{
+			Name: "MY_NODE_NAME",
+			ValueFrom: &v1.EnvVarSource{
+				FieldRef: &v1.ObjectFieldSelector{
+					FieldPath: "spec.nodeName",
+				},
+			},
+		},
+		{
+			Name: "MY_NAMESPACE",
+			ValueFrom: &v1.EnvVarSource{
+				FieldRef: &v1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name:  "VOLUME_CONFIG_NAME",
+			Value: os.Getenv("VOLUME_CONFIG_NAME"),
+		},
+	}
+
+	containers := []v1.Container{
+		{
+			Name:         provisionerContainerName,
+			Image:        defaultImage,
+			VolumeMounts: volumeMounts,
+			Env:          envVars,
+		},
+	}
+
+	// TODO: make daemonset configurable as well, using another configmap.
+	daemonSet := extv1beta1.DaemonSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "extensions/v1beta1",
+			Kind:       "DaemonSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      provisionerDaemonSetName,
+			Namespace: namespace,
+		},
+		Spec: extv1beta1.DaemonSetSpec{
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": provisionerDaemonSetName},
+				},
+				Spec: v1.PodSpec{
+					Volumes:            volumes,
+					Containers:         containers,
+					ServiceAccountName: provisionerServiceAccountName,
+				},
+			},
+		},
+	}
+
+	_, err := client.Extensions().DaemonSets(namespace).Create(&daemonSet)
+	return err
+}
+
+func main() {
+	flag.Set("logtostderr", "true")
+	flag.Parse()
+
+	namespace := os.Getenv("MY_NAMESPACE")
+	if namespace == "" {
+		glog.Fatalf("MY_NAMESPACE environment variable not set\n")
+	}
+	volumeConfigName := os.Getenv("VOLUME_CONFIG_NAME")
+	if volumeConfigName == "" {
+		glog.Fatalf("VOLUME_CONFIG_NAME environment variable not set\n")
+	}
+
+	client := setupClient()
+	config, err := common.GetVolumeConfig(client, namespace, volumeConfigName)
+	if err != nil {
+		glog.Fatalf("Could not get config map information: %v", err)
```

Comment:
if user provides a configmap, then we'll use it and fail if it doesn't exist or is erroneous. Otherwise, we create a configmap with default settings. 

If default configmap already exists, we'll just re-use it provided it is able to be parsed.